app/src/org/commcare/preferences/AdvancedActionsPreferences.java

```diff
@@ -283,11 +284,30 @@ public static void clearUserData(final AppCompatActivity activity) {
             }
             dialog.dismiss();
         };
-        d.setPositiveButton(StringUtils.getStringRobust(activity, R.string.delete_yes), listener);
+        d.setPositiveButton(
+                StringUtils.getStringRobust(activity,
+                        getClearUserDataPositiveOption()),
+                listener);
         d.setNegativeButton(StringUtils.getStringRobust(activity, R.string.cancel), listener);
         d.showNonPersistentDialog();
     }
 
+    private static String getClearUserDataMessage(){
+        int numUnsentAndIncompleteForms = StorageUtils.getNumUnsentAndIncompleteForms();
```

Comment:
we should move this call out of here and `getClearUserDataPositiveOption` and instead pass in this as a function argument. This will avoid having to caclulate this number twice.

Reply:
Agree, I was not happy with that either.